crates/wit-component/tests/components/lower-options/component.wat

```diff
@@ -0,0 +1,284 @@
+(component
+  (type (;0;) (func))
+  (type (;1;) (list string))
+  (type (;2;) (func (param "x" (type 1))))
+  (type (;3;) (record (field "s" string)))
+  (type (;4;) (func (param "x" (type 3))))
+  (type (;5;) (variant (case "s" string)))
+  (type (;6;) (func (param "x" (type 5))))
+  (type (;7;) (record (field "s" u32)))
+  (type (;8;) (func (param "x" (type 7))))
+  (type (;9;) (variant (case "s" u32)))
+  (type (;10;) (func (param "x" (type 9))))
+  (type (;11;) (list (type 3)))
+  (type (;12;) (func (param "x" (type 11))))
+  (type (;13;) (list (type 5)))
+  (type (;14;) (func (param "x" (type 13))))
+  (type (;15;) (list u32))
+  (type (;16;) (func (param "x" (type 15))))
+  (type (;17;) (func (param "x" u32)))
+  (type (;18;) (tuple u32 u32))
+  (type (;19;) (func (result (type 18))))
+  (type (;20;) (func (result string)))
+  (type (;21;) (func (result (type 15))))
+  (type (;22;) (func (result u32)))
+  (type (;23;) (func (result (type 5))))
+  (type (;24;) (list (type 9)))
+  (type (;25;) (func (result (type 24))))
+  (type (;26;) 
+    (instance
+      (alias outer 1 (type (;0;) 0))
```

Comment:
Agreed with all of the above. `wasmprinter` has collections for printing the names of component-y things, but they're all empty with a pending "read an updated names section for components" TODO.

Regarding the confusing comment with type aliasing, we should probably raise that issue with the text format.

Reply:
Hm so [according to this](https://github.com/WebAssembly/component-model/blob/main/design/mvp/Binary.md#alias-definitions) the syntax intended seems to be `(alias outer ct i (type))` so this may also be a mistake in wasmprinter. That being said it's a good idea to open an issue, so I'll open one about switching the order of alias/type

Reply:
Yep, that's a bug in `wasmprinter`. I can get that fixed before merging this so it has the correct baseline.

Reply:
I also opened https://github.com/WebAssembly/component-model/issues/14 for a name section

Reply:
Resolving this comment as we can move forward on the spec issues for now (and also you've fixed the `wasmprinter` bug).